src/pyOpenMS/pyopenms/_dataframes.py

```diff
@@ -608,5 +611,145 @@ def update_scores_from_df(peps: List[_PeptideIdentification], df : _pd.DataFrame
 
     return rets
 
+def _add_meta_values(df: _pd.DataFrame, object: any) -> _pd.DataFrame:
+    """
+    Adds metavalues from given object to given DataFrame.
+    
+    Args:
+        df (pd.DataFrame): DataFrame to which metavalues will be added.
+        obj (any): Object from which metavalues will be extracted.
+    
+    Returns:
+        pd.DataFrame: DataFrame with added meta values.
+    """
+    mvs = []
+    object.getKeys(mvs)
+    for k in mvs:
+        v = object.getMetaValue(k)
+        dtype = 'U100'
+        try:
+            v = int(v)
+            dtype = int
+        except ValueError:
+            try:
+                v = float(v)
+                dtype = 'double'
+            except ValueError:
+                dtype = f'U{len(v)}'
+        
+        df[k.decode()] = _np.full(df.shape[0], v, dtype=_np.dtype(dtype))
+
+    return df
+
+class _MSSpectrumDF(_MSSpectrum):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_df(self, export_meta_values: bool = True, export_peptide_identifications: bool = False) -> _pd.DataFrame:
+        """
+        Returns a DataFrame representation of the MSSpectrum.
+
+        Args:
+            export_meta_values (bool): Whether to export meta values.
+            export_peptide_identifications (bool): Whether to export peptide identifications.
+
+        Returns:
+            pd.DataFrame: DataFrame representation of the MSSpectrum.
+        """
+        def extract_peak_data(s: _MSSpectrum):
```

Comment:
note that we have already some numpy code: https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/addons/MSSpectrum.pyx#L6

Reply:
good point!